backend/internal/auth/sessionmanager.go

```diff
@@ -0,0 +1,50 @@
+package auth
+
+import (
+	"sync"
+	"time"
+
+	"github.com/ibeckermayer/teleport-interview/backend/internal/model"
+)
+
+// Session is an individual user's session
+type Session struct {
+	SessionID SessionID
+	Account   model.Account
+	Expires   time.Time
+}
+
+// SessionManager is an in-memory session store. The app should only ever create one
+// of these and pass it around as a pointer
+type SessionManager struct {
+	store   map[SessionID]Session
+	timeout time.Duration // absolute timeout for individual sessions
+	mtx     sync.RWMutex
+}
+
+// NewSessionManager creates a new *SessionManager
+func NewSessionManager(timeout time.Duration) *SessionManager {
+	return &SessionManager{store: make(map[SessionID]Session), timeout: timeout}
+}
+
+// CreateSession creates a new session in the SessionManager's store, indexed by a
+// new randomly generated SessionID, and expiring sm.timeout from the time it's created.
+// It will return an error if the system's secure random number generator fails to function
+// correctly.
+func (sm *SessionManager) CreateSession(account model.Account) (Session, error) {
+	sid, err := newSessionID()
+	if err != nil {
+		return Session{}, err
+	}
+
+	// TODO: Do I need an RLock here to read sm.timeout?
```

Comment:
you don't need to lock since `sm.timeout` is never modified

Reply:
Got it. My line of thinking was 'well the `sm.store` map might be modified, and that's part of `sm` meaning "`sm` might be modified", so then perhaps I need an `RLock` to read from `sm.timeout`.'

One way I can see for why that must be false is that the reductio-ad-absurdum of those semantics would mean that any object up the hierarchy would need the lock as well (i.e. in this case, the `Server` struct holding the `sm` would need the lock).